### PR TITLE
chore(mutelist): create new class to encapsulate the logic

### PR DESCRIPTION
**Pull request by jfagoagas**
### Description

Encapsulate the `Mutelist` business logic to mark findings as muted to be called as a library within Prowler.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/check/check.py [hunk 1]
```diff
@@ -716,10 +715,29 @@ def execute(
...
+            # TODO: make this prettier
+            if global_provider.type == "aws":
+                is_finding_muted_args["aws_account_id"] = (
+                    global_provider.identity.account
+                )
+            elif global_provider.type == "azure":
+                is_finding_muted_args["subscription"] = global_provider.identity.account
+
+            elif global_provider.type == "gcp":
+                is_finding_muted_args["aws_account_id"] = (
```
Comment (jfagoagas) [inline prowler/lib/check/check.py]: Review this.

File: prowler/lib/check/check.py [hunk 2]
```diff
@@ -716,10 +715,29 @@ def execute(
...
+                    global_provider.identity.account
+                )
+            elif global_provider.type == "kubernetes":
+                is_finding_muted_args["aws_account_id"] = (
```
Comment (jfagoagas) [inline prowler/lib/check/check.py]: Review this.

File: prowler/lib/check/check.py (hunk 1 above)
Comment (MrCloudSec) [inline prowler/lib/check/check.py]:
```suggestion
                is_finding_muted_args["project_id"] = (
```

File: prowler/lib/check/check.py (hunk 2 above)
Comment (MrCloudSec) [inline prowler/lib/check/check.py]:
```suggestion
                is_finding_muted_args["cluster"] = (
```

File: prowler/lib/check/check.py [hunk 3]
```diff
@@ -716,10 +715,29 @@ def execute(
...
+                    global_provider.identity.account
```
Comment (MrCloudSec) [inline prowler/lib/check/check.py]:
```suggestion
                    global_provider.identity.cluster
```

File: prowler/lib/check/check.py [hunk 4]
```diff
@@ -716,10 +715,29 @@ def execute(
...
+            if global_provider.type == "aws":
+                is_finding_muted_args["aws_account_id"] = (
+                    global_provider.identity.account
+                )
+            elif global_provider.type == "azure":
+                is_finding_muted_args["subscription"] = global_provider.identity.account
+
+            elif global_provider.type == "gcp":
+                is_finding_muted_args["aws_account_id"] = (
+                    global_provider.identity.account
```
Comment (MrCloudSec) [inline prowler/lib/check/check.py]: We have to get here the `finding.project_id` value

File: prowler/lib/check/check.py [hunk 5]
```diff
@@ -716,10 +715,29 @@ def execute(
...
-                check_findings,
-            )
+            is_finding_muted_args = {}
+            # TODO: make this prettier
+            if global_provider.type == "aws":
+                is_finding_muted_args["aws_account_id"] = (
+                    global_provider.identity.account
+                )
+            elif global_provider.type == "azure":
+                is_finding_muted_args["subscription"] = global_provider.identity.account
```
Comment (MrCloudSec) [inline prowler/lib/check/check.py]: We have to get here. the `finding.subscription` value.

File: prowler/lib/check/check.py (hunk 4 above)
Comment (MrCloudSec) [reply prowler/lib/check/check.py]: We can get it in the `is_finding_muted` function.

File: prowler/lib/check/check.py (hunk 5 above)
Comment (MrCloudSec) [reply prowler/lib/check/check.py]: We can get it in the `is_finding_muted` function.

File: prowler/providers/azure/lib/mutelist/mutelist.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        self,
+        finding: Any,
+        subscription: str,
+    ) -> bool:
+        return self.is_muted(
+            subscription,
+            finding.check_metadata.CheckID,
+            # TODO: add region to the findings when we add Azure Locations
+            # finding.region,
+            "",
```
Comment (MrCloudSec) [inline prowler/providers/azure/lib/mutelist/mutelist.py]: We do have locations in Azure now :)
```suggestion
            finding.region,
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
This is awesome! Please review my comments 😄

File: prowler/lib/check/check.py (hunk 4 above)
Comment (jfagoagas) [reply prowler/lib/check/check.py]: Great idea!

File: prowler/lib/check/check.py (hunk 5 above)
Comment (jfagoagas) [reply prowler/lib/check/check.py]: Awesome!

File: prowler/providers/azure/lib/mutelist/mutelist.py (hunk 6 above)
Comment (jfagoagas) [reply prowler/providers/azure/lib/mutelist/mutelist.py]: Done but with `finding.location` which is the value for Azure.

**Comment by MrCloudSec** [comment]
Love it @jfagoagas ! Thanks for the changes, once we test it in real time in all the providers, we can merge it.

**Review by MrCloudSec — APPROVED** [review]
Incredible 🏅

**Pull request merged**